### PR TITLE
rubocop: Clean up `Style/BlockDelimiters` excludes and autofix offenses

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- The final part of #14685, maybe?!
- This will seem like a lot, but I feel like `EnforcedStyle: braces_for_chaining` config makes the most sense. It preserves a lot of the RSpec syntax we're used to in `expect {}` blocks that are chained to other things - quite a lot of them - and uses normal block syntax for non-chained methods (as the name suggests - I probably didn't need to write that down).

### CODE REVIEW

File: Library/Homebrew/cask/artifact/abstract_uninstall.rb [hunk 1]
```diff
@@ -152,10 +152,10 @@ def find_launchctl_with_wildcard(search)
         regex = Regexp.escape(search).gsub("\\*", ".*")
         system_command!("/bin/launchctl", args: ["list"])
           .stdout.lines.drop(1) # skip stdout column headers
-          .map do |line|
+          .map { |line|
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/artifact/abstract_uninstall.rb:155]: Sorry, really don't agree with this change. Would rather we enforced specs to use this format than change non-spec code for this.

Comment (issyl0) [reply Library/Homebrew/cask/artifact/abstract_uninstall.rb:155]: So you want `EnforcedStyle: line_count_based` (the default) everywhere (so for multi-line blocks they're always `do ... end`?

Comment (MikeMcQuaid) [reply Library/Homebrew/cask/artifact/abstract_uninstall.rb:155]: @issyl0 Yes I think so.

Comment (issyl0) [reply Library/Homebrew/cask/artifact/abstract_uninstall.rb]: Done!

**Comment by MikeMcQuaid** [comment]
Nice, thanks @issyl0!

**Pull request merged**